### PR TITLE
Require Temurin or Oracle JDK for development

### PR DESCRIPTION
**Pull request by wendigo**
These are distributions that are tested and are known to work fine.

This requirement doesn't mean that Trino won't run on other JDK distributions.

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by losipiuk — APPROVED** [review]
LGTM. @electrum PTAL

File: pom.xml [hunk 1]
```diff
@@ -2579,6 +2579,14 @@
                             <requirePropertyDiverges>
                                 <property>project.description</property>
                             </requirePropertyDiverges>
+                            <requireJavaVendor>
+                                <message>Trino requires Temurin or Oracle JDK for development. Other vendors are not recommended due to lack of testing coverage.</message>
```
Comment (hashhar) [inline pom.xml:2583]: it is "recommended" not required IMO.

People can develop on whatever they want - it'll get tested using Temurin on CI anyway and that's the quality gate.

**Review by hashhar — DISMISSED** [review]

File: pom.xml (hunk 1 above)
Comment (wendigo) [reply pom.xml:2583]: People were reporting some local issues and the reason was always either Corretto or Azul.

Comment (wendigo) [reply pom.xml:2583]: I've discussed that in the past and I believe this is right thing to do.

Comment (electrum) [reply pom.xml:2583]: It seems reasonable if there are weird issues that waste people's time. Better to be strict and fail up front.

**Review by electrum — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (mosabua) [reply pom.xml:2583]: Agreed .. especially also since Temurin is free and open source...

**Pull request merged**